cmake/utils.cmake

```diff
@@ -149,66 +146,23 @@ function(mayaUsd_install_rpath rpathRef NAME)
     )
 endfunction()
 
-function(mayaUsd_promoteMayaUsdHeader)
-    set(srcFile ${CMAKE_CURRENT_SOURCE_DIR}/base/mayaUsd.h.src)
-    set(dstFile ${CMAKE_BINARY_DIR}/include/mayaUsd/mayaUsd.h)
-    if (NOT EXISTS ${dstFile})
-        message(STATUS "promoting: " ${srcFile})
-    endif()
-    configure_file(${srcFile} ${dstFile})
-endfunction()
-
 #
 # mayaUsd_promoteHeaderList(
-#                        [SUBDIR <optional sub-directory>])
-#                        [FILES <list of files>]
+#                        [SUBDIR  <sub-directory name>]
+#                        [FILES   <list of files>]
+#                        [BASEDIR <sub-directory name>])
+#
+#   SUBDIR     - sub-directory in which to promote files.
+#   FILES      - list of files to promote.
+#   BASESDIR   - base dirctory where promoted headers are installed into.
+#                if not defined, mayaUsd subdirectory is used by default. 
 #
-#   SUBDIR        - optional sub-directory in which to promote files.
-#   FILES         - list of files to promote.
 #
 function(mayaUsd_promoteHeaderList)
-    cmake_parse_arguments(PREFIX 
-        ""          # options
-        "SUBDIR"    # one_value keywords
-        "HEADERS"   # multi_value keywords
-        ${ARGN}
-    )
-
-    set(DEST_DIR ${CMAKE_BINARY_DIR}/include/mayaUsd)
-    if(PREFIX_SUBDIR)
-        set(DEST_DIR ${DEST_DIR}/${PREFIX_SUBDIR})
-    endif()
-
-     if(PREFIX_HEADERS)
-        set(headerFiles ${PREFIX_HEADERS})
-    else()
-        message(FATAL_ERROR "HEADERS keyword is not specified.")
-    endif()
-
-    foreach(header ${headerFiles})
-        set(srcFile ${CMAKE_CURRENT_SOURCE_DIR}/${header})
-        set(dstFile ${DEST_DIR}/${header})
-
-        set(content "#pragma once\n#include \"${srcFile}\"\n")
-
-        if (NOT EXISTS ${dstFile})
-            message(STATUS "promoting: " ${srcFile})
-            file(WRITE ${dstFile} "${content}")
-        else()
-            file(READ ${dstFile} oldContent)
-            if (NOT "${content}" STREQUAL "${oldContent}")
-                message(STATUS "Promoting ${srcfile}")
-                file(WRITE ${dstFile} "${content}")
-            endif()
-        endif()
-    endforeach()
-endfunction()
-
-function(mayaUsd_promoteHeaderListWithSubdir)
     cmake_parse_arguments(PREFIX
         ""
         "SUBDIR" # one_value keywords
-        "HEADERS;BASE_PATH" # multi_value keywords
+        "HEADERS;BASEDIR" # multi_value keywords
```

Comment:
I know you're not changing this, but... why is BASEDIR a multi-value keyword?  The usage of `${PREFIX_BASEDIR}` below doesn't seem to indicate it would work well with multiple items.

Reply:
Yup, this happens when one copy pastes stuff around!!! BASEDIR  is a one-value keyword. Thx.
See d470ae2

Reply:
Cool, thanks!